pkgs/applications/radio/sdrplay/default.nix

```
@@ -0,0 +1,46 @@
+{ stdenv, lib, fetchurl, autoPatchelfHook, libudev }:
+let
+  arch = if stdenv.isx86_64  then "x86_64"
+    else if stdenv.isi686    then "i686"
+    else throw "unsupported architecture";
+in
+stdenv.mkDerivation rec {
+  name = "sdrplay";
+  version = "3.07.1";
+
+  # soapysdrplay also needs to full so name
+  soName = "libsdrplay_api.so." + lib.concatStringsSep "." (lib.take 2 (builtins.splitVersion version));
```

```suggestion
  passthru.soName = "libsdrplay_api.so." + lib.concatStringsSep "." (lib.take 2 (builtins.splitVersion version));
```
You are also using it in the soapysdr module. I think this should be a `passthru` property.

---

Taking some inspiration from the sdrplay2 implementation of @lasandell i was able elide this property completely :smile: 
I have yet to understand how this pkg-config automagic works though.